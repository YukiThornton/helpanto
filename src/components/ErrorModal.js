import React from 'react';
import Modal from 'react-modal';
import FlatButton from 'material-ui/FlatButton';
import PropTypes from 'prop-types';

const ErrorModal = ({isOpen, message, onRequestClose}) => (
  <Modal
    isOpen={isOpen}
    onRequestClose={onRequestClose}
  >
    <h1>Error</h1>
    <p>{message}</p>
    <FlatButton
      onClick={onRequestClose}
      label="Close"
      fullWidth={true}
    />
  </Modal>
);

ErrorModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  message: PropTypes.string,
  onRequestClose: PropTypes.func.isRequired,
};

export default ErrorModal;
