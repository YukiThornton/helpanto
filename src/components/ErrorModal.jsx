import React from 'react';
import Modal from 'react-modal';
import PropTypes from 'prop-types';
import isRequiredIf from 'react-proptype-conditional-require';
import FlatButton from 'material-ui/FlatButton';

const ErrorModal = ({
  isOpen,
  message,
  onRequestClose,
}) => (
  <Modal
    isOpen={isOpen}
    onRequestClose={onRequestClose}
  >
    <h1>Error</h1>
    <p>{message}</p>
    <FlatButton
      onClick={onRequestClose}
      label="Close"
      fullWidth
    />
  </Modal>
);

ErrorModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  message: isRequiredIf(PropTypes.string, props => props.isOpen),
  onRequestClose: PropTypes.func.isRequired,
};

export default ErrorModal;
