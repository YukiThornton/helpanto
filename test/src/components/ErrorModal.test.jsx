import React from 'react';
import { shallow } from 'enzyme';
import ErrorModal from '../../../src/components/ErrorModal';

describe('test ErrorModal', () => {
  it('should render no modal when isOpen === false', () => {
    const component = (
      <ErrorModal
        isOpen={false}
        onRequestClose={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });

  it('should render a modal when isOpen === true', () => {
    const component = (
      <ErrorModal
        isOpen
        message="Error message"
        onRequestClose={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });
});
