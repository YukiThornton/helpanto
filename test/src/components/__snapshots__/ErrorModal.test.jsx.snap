// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorModal should render a modal when isOpen === true 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Error
  </h1>
  <p>
    Error message
  </p>
  <FlatButton
    disabled={false}
    fullWidth={true}
    label="Close"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
  />
</Modal>
`;

exports[`test ErrorModal should render no modal when isOpen === false 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1>
    Error
  </h1>
  <p />
  <FlatButton
    disabled={false}
    fullWidth={true}
    label="Close"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
  />
</Modal>
`;
